### PR TITLE
[SPARK-43895][CONNECT][GO] Prepare the go package path

### PR DESCRIPTION
**Pull request by grundprinzip**
### What changes were proposed in this pull request?

This patch adds the golang package path to the proto files to be consumed from the golang repo.

### Why are the changes needed?
Preparation for the go client.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
No functional change.

### CODE REVIEW

**Review by hvanhovell — APPROVED** [review]
LGTM

**Comment by grundprinzip** [comment]
This trips the breaking change detection because the previous go package path was empty. However, this is not an issue because we don't have a go client as of right now.

image

File: connector/connect/common/src/main/protobuf/buf.yaml [hunk 1]
```diff
@@ -18,6 +18,8 @@ version: v1
 breaking:
   use:
     - FILE
+  except:
+    - FILE_SAME_GO_PACKAGE
```
[the remark this answers is not in the document]
Comment (amaliujia) [reply connector/connect/common/src/main/protobuf/buf.yaml:22]: Ah I see this is to compare with previous release. Thanks for the clarification.

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
Thanks, merging to master

**Pull request closed without merging**